### PR TITLE
Update Signer

### PR DESCRIPTION
**Pull request by sesheffield**
### Description
* added readme
* remove kava/app dependency
* fixed memory leaks and error handling
* refactors
* gracefully shutdown tests

### Testing:
* Ran `testing/main.go` before some refactoring, producing 1000 messages sent and 1000 transactions.
* Reran `testing/main.go` afte refactoring, producing 1000 messages sent and 1000 transactions.

### Related Tickets
* [Docs Ticket](https://app.shortcut.com/kava-labs/story/2511/add-basic-documentation-for-go-tools-signer)
* [Refactor Ticket](https://app.shortcut.com/kava-labs/story/2510/remove-kava-app-dependency-in-go-tools-signer)

### CODE REVIEW

File: signing/README.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    - Handles various errors and retries broadcasting as needed. 
+    - Manages sequence errors and adjusts the transaction sequence to ensure transactions are broadcast correctly.
+
+- **Sign**:
+  - Signs a transaction using the provided private key and signer data, returns the signed transaction and its raw bytes.
+  
+- **GetAccAddress**:
+  - Returns the account address for a given private key.
+
+## Broadcast Loop Logic:
```
Comment (karzak) [inline signing/README.md:50]: Can we add a section on running the tests? It looks like `go test ./...` isn't the expected workflow, so good to document the exact steps (ie run a local chain with kvtool, `go run .` from the testing folder).

**Review by karzak — COMMENTED** [review]
Great work, thanks for the much needed cleanups and improvements!

**Review by karzak — APPROVED** [review]
LGTM

File: signing/testing/main.go [hunk 2]
```diff
@@ -28,78 +28,118 @@ const (
...
 		}
 
-		fmt.Printf("sent msg %d\n", i+1)
+		logger.Info().Int("msg", i+1).Msg("Sent message")
 	}
 
-	for {
-	}
+	// Block indefinitely to keep the program running.
+	select {}
```
Comment (pirtleshell) [inline signing/testing/main.go:144]: lol, nice catch

File: signing/encoding-config.go [hunk 3]
```diff
@@ -0,0 +1,28 @@
+package signing
```
Comment (pirtleshell) [inline signing/encoding-config.go:1]: hyphens should be avoided in the filename here. prefer either `encodingconfig`, `encoding_config`, or just `encoding`

but question: what is the motivation behind this adapter? it seems unused besides being a passthrough for the sdk codec. i'd omit it and avoid the added complexity

File: signing/signer.go [hunk 4]
```diff
@@ -78,71 +92,110 @@ func NewSigner(
...
 }
 
-func (s *Signer) pollAccountState() <-chan authtypes.AccountI {
+func (s *Signer) setAccountError(err error) {
+	s.accStatus = err
+}
+
+func (s *Signer) clearAccountError() {
+	s.accStatus = nil
+}
```
Comment (pirtleshell) [inline signing/signer.go:109]: not sure these methods make the code easier or harder to read. personally, i prefer to not have to jump to another method signature to know that setting the error just sets a private variable.

**Review by pirtleshell — CHANGES_REQUESTED** [review]
awesome job improving the signer's documentation and logging 🏆 

i would love if you could pull those changes into a separate PR so we can discuss the functional (and breaking) changes in the signer method signatures & how shutdowns are handled, and include them separately.

i have more to say about the support of shutdowns / cancellation with the use of `ctx.Done()` than my comments here. 

my current feedback is that you are doing too much in this PR, especially considering it has implications in two of our production services, which will require deployments post-merge. the more we change (however improved or more idiomatic), the more things can go wrong, so we need to consider the tradeoff. please break this pr up into smaller changesets.

File: signing/encoding-config.go (hunk 3 above)
Comment (sesheffield) [reply signing/encoding-config.go:1]: https://app.shortcut.com/kava-labs/story/2510/remove-kava-app-dependency-in-go-tools-signer

**Pull request closed without merging**